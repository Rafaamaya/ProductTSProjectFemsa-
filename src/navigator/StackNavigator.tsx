
import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import {  DetailsScreen } from '../screens/DetailScreen/DetailsScreen';
import HomeScreen from '../screens/HomeScreen/HomeScreen';
import { Product } from '../interfaces/productInterface';
export type RootStackParams = {
    HomeScreen: undefined,
    DetailsScreen: { item: Product },
}

const Stack = createStackNavigator<RootStackParams>();

export const StackNavigator = () => {
    return (
        <Stack.Navigator
            screenOptions={{
               headerShown:false,
            }}
        >
            <Stack.Screen
                name="HomeScreen"
                options={{ title: '' }}
                component={HomeScreen}
            />
            <Stack.Screen
                name="DetailsScreen"
                options={{ title: 'Clima' }}
                component={DetailsScreen}
            />
        </Stack.Navigator>
    );
};
