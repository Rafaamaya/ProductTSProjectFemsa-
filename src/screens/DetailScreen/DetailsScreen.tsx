import React from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import {RootStackParams} from '../../navigator/StackNavigator';
import {StackNavigationProp, StackScreenProps} from '@react-navigation/stack';
import {formatFeedDate} from '../../utils/timeUtils';
import { useNavigation } from '@react-navigation/native';
import styles from './styles';

interface Props extends StackScreenProps<RootStackParams, 'DetailsScreen'> {}
type HomeScreenProp = StackNavigationProp<RootStackParams, 'HomeScreen'>;


export const DetailsScreen = ({route}: Props) => {
  const {item} = route.params;
  const navigation = useNavigation<HomeScreenProp>();

  return (
    <>
      <View style={styles.hederContainer}>
        <Text style={styles.title}>{item.product}</Text>
      </View>
      <View style={styles.container}>
        <View style={styles.imageContainer}>
          <Image style={styles.image} source={{uri: item.image}} />
        </View>
        <Text style={styles.descriptiveTitle}>Detalles del producto:</Text>
        <Text style={styles.textDate}>
          Comprado el {formatFeedDate(item.createdAt)}
        </Text>
        <Text style={styles.descriptiveTitle}>Con esta compra acumulaste:</Text>
        <Text style={styles.textPoints}>{item.points} puntos</Text>
        <TouchableOpacity style={styles.containerButton} onPress={() => navigation.goBack()}>
          <Text style={styles.textButton}>Aceptar</Text>
        </TouchableOpacity>
      </View>
    </>
  );
};


