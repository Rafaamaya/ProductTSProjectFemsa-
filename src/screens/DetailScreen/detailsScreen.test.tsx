import React from 'react';
import renderer from 'react-test-renderer';
import {DetailsScreen} from './DetailsScreen';
import {StackScreenProps} from '@react-navigation/stack/lib/typescript/src/types';
import {RootStackParams} from '../../navigator/StackNavigator';

const MOCKED_ROUTE = {
  params: {
    item: {
      createdAt: '2022-12-09T06:34:25.607Z',
      product: 'Handmade Metal Shoes',
      points: 16434,
      image: 'https://loremflickr.com/640/480/technics',
      isRedemption: false,
      id: '1',
    },
  },
};

export type TestNavigationProps = StackScreenProps<
  RootStackParams,
  'DetailsScreen'
>;
const navigation: Partial<TestNavigationProps> = {};
describe('Pruebas en <DetailsScreen/>', () => {
  test('debe de hacer match con el snapchot', () => {
    const tree = renderer
      .create(
        <DetailsScreen
          navigation={navigation as any}
          route={MOCKED_ROUTE as any}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
