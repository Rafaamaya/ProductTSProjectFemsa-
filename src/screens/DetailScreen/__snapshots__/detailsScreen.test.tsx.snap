// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DetailsScreen/> debe de hacer match con el snapchot 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#CFD6FF",
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Avenir",
          "fontSize": 24,
          "fontWeight": "800",
          "marginBottom": 24,
          "marginLeft": 20,
          "marginTop": 49,
        }
      }
    >
      Handmade Metal Shoes
    </Text>
  </View>,
  <View
    style={
      Object {
        "marginHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://loremflickr.com/640/480/technics",
          }
        }
        style={
          Object {
            "borderRadius": 10,
            "height": 350,
            "marginTop": 10,
            "width": 353,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#9B9898",
          "fontFamily": "Avenir",
          "fontSize": 14,
          "fontWeight": "800",
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      Detalles del producto:
    </Text>
    <Text
      style={
        Object {
          "color": "#020202",
          "fontFamily": "Avenir",
          "fontSize": 16,
          "fontWeight": "800",
        }
      }
    >
      Comprado el 
      9 de Diciembre, 2022
    </Text>
    <Text
      style={
        Object {
          "color": "#9B9898",
          "fontFamily": "Avenir",
          "fontSize": 14,
          "fontWeight": "800",
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      Con esta compra acumulaste:
    </Text>
    <Text
      style={
        Object {
          "color": "#020202",
          "fontFamily": "Avenir",
          "fontSize": 24,
          "fontWeight": "800",
        }
      }
    >
      16434
       puntos
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#334FFA",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 31,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir",
            "fontSize": 16,
            "fontWeight": "800",
          }
        }
      >
        Aceptar
      </Text>
    </View>
  </View>,
]
`;
